pkg/api/avatar/avatar.go

```diff
@@ -104,8 +105,8 @@ func (this *CacheServer) Handler(ctx *macaron.Context) {
 
 	if avatar.notFound {
```

Comment:
@marefr I reviewed the logic carefully now, and reached the conclusion that `avatar` only gets added to the cache if it was found when consulting the server (the server gets consulted for expired avatars in the cache and avatars not found in cache) and not already in cache.

If the existing logic (described above) is correct, I'm not sure there's so much we can do to improve the code here. I'd be interested in your thoughts on this.